ksp/build.gradle.kts

```diff
@@ -5,19 +5,45 @@ plugins {
   alias(libs.plugins.mavenPublish)
 }
 
-tasks.withType<KotlinCompile>()
+tasks
+  .withType<KotlinCompile>()
   .matching { it.name.contains("test", ignoreCase = true) }
   .configureEach {
-    compilerOptions {
-      freeCompilerArgs.add("-opt-in=org.jetbrains.kotlin.compiler.plugin.ExperimentalCompilerApi")
-    }
+    compilerOptions { optIn.add("org.jetbrains.kotlin.compiler.plugin.ExperimentalCompilerApi") }
   }
 
+// It's not possible to test both KSP 1 and KSP 2 in the same compilation unit
+val testKsp2 = providers.systemProperty("kct.test.useKsp2").getOrElse("false").toBoolean()
+
+tasks.test {
+  if (testKsp2) {
+    systemProperty("kct.test.useKsp2", testKsp2)
+  }
+}
+
 dependencies {
   compileOnly(libs.kotlin.compilerEmbeddable)
   api(projects.core)
+
+  // KSP's AA dependencies point to the wrong common-deps artifact
+  compileOnly(libs.ksp.aa) { exclude(group = "com.google.devtools.ksp", module = "common-deps") }
+  compileOnly(libs.ksp.commonDeps)
   compileOnly(libs.ksp.api)
-  implementation(libs.ksp)
+  compileOnly(libs.ksp)
+
+  if (testKsp2) {
```

Comment:
If people don't mind the ~60MB download size, KSP2 should be able to coexist with KSP1 in classpath, with just the following dependencies:
1. `implementation(libs.ksp.aa.embeddable)` // KSP2
2. `implementation(libs.ksp.commonDeps)` // KSP2
3. `implementation(libs.ksp)` // the existing KSP1

Reply:
See https://github.com/ZacSweers/kotlin-compile-testing/pull/196/#discussion_r1480904613